### ISSUE TITLE
job_queue.run_daily fails for timezone aware datetime.datetime objects

### ISSUE THREAD

**Issue by Tianmaru** [python-telegram-bot/python-telegram-bot#1409]
job_queue.run_daily fails for timezone aware datetime.datetime objects (version 12.0.0b1).
It would be nice if you could give timezone-information, for example, if the bot is being hosted in another timezone.

**Comment by torresjrjr** [comment]
I have the same problem. My testing is local at BST (British Summer Time) but my hosting uses UTC time. I think personally the wrapper should use `utcnow()` instead of `now()` when calculating timedeltas for the job queue, because it avoids ambiguous code.

Otherwise, I also wish jobqueue.run_once(when=***) could accept timezone/utc_offset-aware datetime objects too.

**Comment by torresjrjr** [comment]
A solution in the case for `jobqueue.run_once()` is:
```
timedelta_obj = datetime_obj - datetime.datetime.utcnow()
jobqueue.run_once(callback, timedelta_obj)
```
The issue still stands for `jobqueue.run_daily`.

**Comment by Eldinnie** [comment]
As we mentioned on the linked closed PR.
Servers should run on UTC, calculations to different timezones should be the responsibility of the user. closing this.

**Comment by Tianmaru** [comment]
> As we mentioned on the linked closed PR.
> Servers should run on UTC, calculations to different timezones should be the responsibility of the user. closing this.

It would be a clean and easy solution if the user has just to calculate the UTC offset, but it can become very cumbersome for timezones with Daylight Saving Time. If you want a job to run daily on a specific time, you have to restart `jobqueue.run_daily` atleast two times a year with different UTC-timestamps. The use of timezone aware datetime objects would simplify this a lot.

**Comment by Bibo-Joshi** [comment]
@Tianmaru Regarding DST: This was "fixed" by documentation, see #1381
